sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -4713,78 +4782,92 @@ SIREPO.app.directive('sbatchLoginModal', function() {
+		    return;
+		}
+		loginSuccess = true;
                 el.modal('hide');
             }
 
-            $scope.$on('showSbatchLoginModal', function(e, broadcastArg) {
-                // When a user enters invalid login creds 'showSbatchLoginModal' is
-                // broadcast again. onHidden keeps a references to the
-                // errorCallback of only the first broadcast's errorCallback
-                const p = broadcastArg.srExceptionParams;
-                if (onHidden === null) {
-                    onHidden = broadcastArg.errorCallback;
-                }
-                sbatchLoginStatusService.loggedIn = false;
-                $scope.otp = '';
-                $scope.password = '';
-                awaitingSendResponse = false;
-                $scope.host = p.host;
-                $scope.showOtp = p.host.indexOf('nersc') >= 0;
-                $scope.showWarning = p.reason === 'invalid-creds';
-                $scope.warningText = 'Your credentials were invalid. Please try again.';
+	    function resetLoginForm() {
+		resetLoginFormTextFields();
+		$scope.sbatchLoginModalForm.$setPristine();
+                awaitingLoginResponse = false;
+	    }
+
+	    function resetLoginFormTextFields() {
+		$scope.otp = '';
+		$scope.password = '';
+		$scope.username = '';
+	    }
+
+            $scope.$on('showSbatchLoginModal', function(event, broadcastArg) {
+		restartRunSimulation = broadcastArg.shouldRestartRunSimulation;
+		resetLoginForm();
+                $scope.showOtp = authState.sbatchHostIsNersc;
+		$scope.warningText = {
+		    'invalid-creds': 'Your credentials were invalid. Please try again.',
+		    'general-connection-error': `There was a problem connecting to ${$scope.sbatchHostDisplayName}. Please try again. If the issue persists contact support@sirepo.com.`,
+		}[broadcastArg.reason] || '';
                 $scope.submit = function() {
-                    awaitingSendResponse = true;
+                    awaitingLoginResponse = true;
                     requestSender.sendRequest(
                         'sbatchLogin',
-                        handleResponse,
-                        {
-                            otp: $scope.otp,
-                            password: $scope.password,
-                            computeModel: p.computeModel,
-                            simulationId: p.simulationId,
-                            simulationType: SIREPO.APP_SCHEMA.simulationType,
-                            username: $scope.username,
-                        },
-                        handleResponse
+			handleLoginResponse,
+			{
+			    computeModel: broadcastArg.computeModel,
+			    simulationId: broadcastArg.simulationId,
+			    simulationType: SIREPO.APP_SCHEMA.simulationType,
+			    sbatchCredentials: {
+				otp: $scope.otp,
+				password: $scope.password,
+				username: $scope.username,
+			    },
+			    // Handles the case where user clicked on run simulation,
```

Comment:
Comments indicate a need for a function. Make the function which adds the parameter only if it should be added.

Reply:
And eliminate the comment? I'm fine making a function but the comment is useful. This code took me a while to wrap my mind around. fe14ec853

Reply:
Leave the comment, but that's at the top of the function instead of in the middle of the code.